plugins/VisitsSummary/Controller.php

```diff
@@ -103,6 +104,12 @@ public function getEvolutionGraph()
             'avg_time_generation'
         );
 
+        $currentPeriod = Common::getRequestVar('period', false);
+
+        if (!SettingsPiwik::isUniqueVisitorsEnabled($currentPeriod)) {
+            $selectableColumns = array_diff($selectableColumns, ['nb_uniq_visitors']);
```

Comment:
does it also need to remove nb_users?

Reply:
I'm not sure but there might be a method to get the unique metrics?

Reply:
true, `nb_users` needs to be removed as well. But I didn't find any method returning unique metrics...

Reply:
Should we maybe add one? Makes it just easier in the future... seeing similar behaviour in `VisitsSummary\Api` etc and be maybe good to no longer hard code these metrics

Reply:
Actually I'm not sure if that would the optimal solution. I would prefer having a Metric class for each available metric, so we could use a `isAvailable` method to check each metric. Also that would make it easier to get the translation, documentation and stuff like that for each metric.
But that's a bit a bigger task we could maybe do for Matomo 4.